Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -772,7 +772,9 @@ extension TaskHandler: ChannelDuplexHandler {
         var headers = request.headers
 
         if !request.headers.contains(name: "Host") {
-            headers.add(name: "Host", value: request.host)
+            let port = request.port
+            let host = (port != 80 && port != 443) ? "\(request.host):\(port)" : request.host
+            headers.add(name: "Host", value: host)
```

Comment:
can we do lower-cased header names? That simplifies HTTP/2 because the 2to1 handler should then be able to just pass it through (yes, `Host` becomes `:authority` but I think we should do lowercase only).

Reply:
done